devito/finite_differences/lazy.py

```diff
@@ -47,3 +49,15 @@ def evaluate(self):
         args = self._evaluate_args()
         evaluate = not all(i is j for i, j in zip(args, self.args))
         return self.func(*args, evaluate=evaluate)
+
+
+# Custom SymPy types used upon evaluation
+
+
+class EvalDerivative(sympy.Add):
```

Comment:
Why is it a sympy Add and not ours?

Reply:
Because EvalDerivative is what we get upon evaluation. And by the time evaluation is performed, all the FD types will have been lowered to pure SymPy ones (ie, inside the compiler after Equation lowering we enforce pure SymPy for homogeneity)

Reply:
Inside the compiler sure, but evaluation is still a user level feature and it's output (the evaluated expression) should still be differentiable

Reply:
> it's output (the evaluated expression) should still be differentiable

they are indeed because upon .evaluate you get objects of type `EvalDiffDerivative` (which is part of the `Differentiable` hierarchy) while _only inside the compiler_, I think within `LoweredEq(...)`, you get `EvalDerivative`

also, all tests are passing so we should be good on this front

Reply:
>.evaluate you get objects of type EvalDiffDerivative 

Right ok should be good then indeed.